docs/library/graphing/plotly.md

```diff
@@ -10,104 +10,144 @@ import reflex as rx
+            x="year",
+            y="lifeExp",
+            title=f"Life expectancy in {country}",
+        )
 
-Now pass the plotly figure to the Plotly component.
 
-```python
-rx.plotly(data=fig, height="400px")
+def line_chart_with_state():
+    return rx.vstack(
+        rx.select(
+            ['China', 'France', 'United Kingdom', 'United States', 'Canada'],
+            default_value="United States",
+            on_change=PlotlyState.set_selected_country,
+        ),
+        rx.plotly(data=PlotlyState.figure),
+    )
 ```
 
-Now lets take a look at a more complex example.
-Let's create a 3D surface plot of Mount Bruno.
+## Adding Styles and Layouts
+Use `update_layout()` method to update the layout of your chart. Checkout [Plotly Layouts](https://plotly.com/python/reference/layout/) for all layouts props. 
 
-```python eval
-rx.center(
-  rx.vstack(
-      rx.plotly(data=mount()),
-  ),
-  height="400px",
-  style=style["box"],
-)
+```md alert info
+Note that the width and height props are not recommended to ensure the plot remains size responsive to its container. 
 ```
 
-Read in the Mount Bruno data as a csv and create a plotly figure.
-
-```python
-import plotly.graph_objects as go
+```python demo exec
+df = px.data.gapminder().query("country=='Canada'")
+fig_1 = px.line(
+    df,
+    x="year",
+    y="lifeExp",
+    title="Life expectancy in Canada",
+)
+fig_1.update_layout(
+    title_font_size = 25,
+    title_x = 0.5, # at the center",
+    plot_bgcolor = "#c3d7f7",
+    paper_bgcolor = "#e0e0e0",
+)
+def add_styles():
+    return rx.center(
+        rx.plotly(data=fig_1),
+    )
+```
 
-import pandas as pd
+## Dimension and Resizing
+The auto-resizing is enabled by default (set to True). The size of the plot will be responsive according to the screen size. To disable auto-resizing, you need to set `use_resize_handler` to False. You can size your Plotly chart using the width and height props, just like any other Reflex component.
 
-# Read data from a csv
-z_data = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/api_docs/mt_bruno_elevation.csv')
+```md alert info
+It is recommended to set the width and height in the `rx.plotly` component instead of using the `update_layout()` method. The latter will disable auto-resizing.
+```
 
-fig = go.Figure(data=[go.Surface(z=z_data.values)])
-fig.update_traces(contours_z=dict(show=True, usecolormap=True,
-                                  highlightcolor="limegreen", project_z=True))
-fig.update_layout(scene_camera_eye=dict(x=1.87, y=0.88, z=-0.64),
-                  width=500, height=500,
-                  margin=dict(l=65, r=50, b=65, t=90)
+```python demo exec
+df = px.data.gapminder().query("country=='Canada'")
+fig_1 = px.line(
+    df,
+    x="year",
+    y="lifeExp",
+    title="Life expectancy in Canada",
+)
+fig_1.update_layout(
+    title_font_size = 25,
+    title_x = 0.5,
+    plot_bgcolor = "#c3d7f7",
+    paper_bgcolor = "#e0e0e0",
 )
+def resizing():
```

Comment:
This resize example does not actually seem to resize when I change the size of the screen (at least it does not in safari)